partiql-lang/src/main/kotlin/org/partiql/lang/eval/physical/PhysicalPlanCompilerImpl.kt

```diff
@@ -1886,14 +1885,6 @@ internal class PhysicalPlanCompilerImpl(
         )
 }
 
-internal val MetaContainer.sourceLocationMeta get() = this[SourceLocationMeta.TAG] as? SourceLocationMeta
-internal val MetaContainer.sourceLocationMetaOrUnknown get() = this.sourceLocationMeta ?: UNKNOWN_SOURCE_LOCATION
-
-internal fun StaticType.getTypes() = when (val flattened = this.flatten()) {
-    is AnyOfType -> flattened.types
-    else -> listOf(this)
-}
```

Comment:
When I remove these top-level internal functions, this is a breaking change from Java. From the Java to Kotlin interop [docs](https://kotlinlang.org/docs/java-to-kotlin-interop.html#visibility):

> `internal` declarations become `public` in Java. Members of `internal` classes go through name mangling, to make it harder to accidentally use them from Java and to allow overloading for members with the same signature that don't see each other according to Kotlin rules

So from Java code, `internal` classes will have some name mangling. But for top-level `internal` functions, there is no name mangling. For instance, `StaticType.getTypes()` is directly callable from Java by: 
```
PhysicalPlanCompilerAsyncImplKt.getTypes(StaticType.ANY);
```

IntelliJ will give an error like "Usage of Kotlin internal declaration from different module " but these are just errors in the IDE that can be disabled. Gradle and the Java compiler can still run the internal Kotlin code.

I'll cut an issue to track if we should allow for these top-level functions (both `internal` and `public`).

Reply:
GitHub issue: https://github.com/partiql/partiql-lang-kotlin/issues/1387